packages/netlify-cms-core/src/reducers/collections.ts

```diff
@@ -245,6 +245,21 @@ export const traverseFields = (
   if (done()) {
     return fields;
   }
+
+  // List of widgets and options that are still using camelCase
+  const widgetArray = ['date', 'datetime', 'markdown', 'number', 'relation'];
```

Comment:
We should not include specific widget names in the core.
Regardless this code should be done in:
https://github.com/netlify/netlify-cms/blob/331f4b6ac171eb9e6196a29fa8c8371cd7846d04/packages/netlify-cms-core/src/actions/config.js#L286

Also, we would need to update the places in the code that access those fields.
For example:
https://github.com/netlify/netlify-cms/blob/331f4b6ac171eb9e6196a29fa8c8371cd7846d04/packages/netlify-cms-widget-datetime/src/DateTimeControl.js#L27

We can use a similar approach to:
https://github.com/netlify/netlify-cms/blob/331f4b6ac171eb9e6196a29fa8c8371cd7846d04/packages/netlify-cms-core/src/actions/config.js#L83
To update all fields